### PR TITLE
fix: validate compatibility between connector and row format

### LINKED ISSUE #6984: validate compatibility between connector type and row format
For example:
```
create materialized source s1 (
    v1 int, 
    v2 float,
    primary key (v1)
) with (
    connector = 'datagen',
    fields.v1.kind = 'sequence',
    fields.v1.start = '1',
    fields.v1.end  = '10',
    fields.v2.kind = 'sequence',
    fields.v2.start = '11',
    fields.v2.end = '20',
    datagen.rows.per.second='15',
    datagen.split.num = '1'
) row format debezium_json;
```

The `create source` succeeds but warning for every row in the log.

Probably validating as early as possible is better?
Or the current approach is intentional?

It's a two-dimensional matrix, isn't it? 
connector type X row format type.

### PR DESCRIPTION
**Pull request by waruto210**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- validate compatibility between connector and row format. fix #6984

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~
~- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Connector (sources & sinks)
- SQL commands, functions, and operators

### Release note

</details>

connectors and their supported for formats:
- kafka: json, protobuf, avro, debezium_json, debezium_avro, maxwell, canal_json, upsert_json, upsert_avro
- pulsar: json, protobuf, avro, debezium_json, maxwell, canal_json
- kinesis: json, protobuf, avro, debezium_json, maxwell, canal_json
- google_pubsub: json, protobuf, avro, debezium_json, maxwell, canal_json
- nexmark: None(internal native format)
- datagen: json or None(internal native format)
- s3: json(nd_json), csv
- mysql_cdc: debezium_json
- postgresql_cdc: debezium_json

Also, user can only use confluent schema regsitry with kafka connecotr.

### CODE REVIEW

File: e2e_test/ddl/invalid_operation.slt [hunk 1]
```diff
@@ -361,6 +361,17 @@ create table sink(v int);
 statement error view.*exists
 create table v(v int);
 
+# 8. create with unsupport format
+statement error
```
Comment (lmatz) [inline e2e_test/ddl/invalid_operation.slt:365]: sqllogictest supports testing whether the error message matches what we expect

**Review by lmatz — APPROVED** [review]
LGTM

**Review by tabVersion — APPROVED** [review]
rest LGTM

**Comment by tabVersion** [comment]
please summarize connector and format compatibilities in pr. user should be aware of this

File: e2e_test/source/basic/kafka.slt [hunk 2]
```diff
@@ -244,6 +244,18 @@ create source s18 with (
...
+# we cannot use confluent schema registry when connector is not kafka
+statement error
+create table s19 
+with (
+  connector = 'kafka',
+  topic = 'topic',
+  properties.bootstrap.server = '127.0.0.1:29092'
+)
+row format avro
+row schema location confluent schema registry 'http://127.0.0.1:8081'
```
[the remark this answers is not in the document]
Comment (waruto210) [reply e2e_test/source/basic/kafka.slt]: > why we reject this case

Sorry, it's a bug.

**Pull request merged**